frame/nomination-pools/src/lib.rs

```diff
@@ -2523,6 +2943,52 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	fn do_claim_commission(who: T::AccountId, pool_id: PoolId) -> DispatchResult {
+		let bonded_pool = BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
```

Comment:
This check could probably be moved into the dispatchable, as in `fn set_commission_*`.